src/Dynatrace.OpenTelemetry.Exporter.Metrics/DynatraceMetricSerializer.cs

```diff
@@ -111,106 +148,80 @@ private string FormatDouble(double min)
 
         private void WriteTimestamp(StringBuilder sb, DateTime timestamp)
         {
-            sb.Append($" {new DateTimeOffset(timestamp).ToUniversalTime().ToUnixTimeMilliseconds()}");
+            sb.Append($" {new DateTimeOffset(timestamp.ToLocalTime()).ToUnixTimeMilliseconds()}");
         }
 
         private void WriteSum(StringBuilder sb, double sumValue)
         {
             sb.Append($" count,delta={sumValue}");
         }
 
-        private void WriteMetricKey(StringBuilder sb, Metric metric)
+        /// <summary>
+        /// Transforms OpenTelemetry metric names to metric keys valid in Dynatrace
+        /// </summary>
+        /// <returns>a valid metric key or null, if the input could not be normalized</returns>
+        private string CreateMetricKey(Metric metric)
         {
             var keyBuilder = new StringBuilder();
             if (!string.IsNullOrEmpty(_prefix)) keyBuilder.Append($"{_prefix}.");
+            // todo is this needed?
             if (!string.IsNullOrEmpty(metric.MetricNamespace)) keyBuilder.Append($"{metric.MetricNamespace}.");
             keyBuilder.Append(metric.MetricName);
-            sb.Append(ToMintMetricKey(keyBuilder.ToString()));
+            return Normalize.MetricKey(keyBuilder.ToString());
         }
 
-        private void WriteDimensions(StringBuilder sb, IEnumerable<KeyValuePair<string, string>> labels)
+        // Items from Enumerables passed further right overwrite items from Enumerables passed further left.
+        // Pass only normalized dimension lists to this function.
+        // The order of the items passed will be preserved.
```

Comment:
That's not necessary since the order carries no certain meaning.
You should be able to just `foreach list { foreach item { dict[key] = value } }` in the order lists are passed.

Reply:
I agree, but I remember we decided to use an ordered dictionary anyway, not entirely sure why though anymore. It certainly makes testing easier, but I think we had a "real" reason too.

Reply:
The "real" reason might be that doing it this way we can guarantee that creating the same metric twice will actually result in the same list of dimensions (i.e. if we use the name and dimensions of the metric as identity).

Reply:
Alright let's see if we need the deterministic order for delta conversion then and if not we can still kick it out to simplify it.